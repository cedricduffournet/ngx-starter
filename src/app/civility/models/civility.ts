import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
