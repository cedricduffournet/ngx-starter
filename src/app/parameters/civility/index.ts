export * from './parameter-civility.component';
