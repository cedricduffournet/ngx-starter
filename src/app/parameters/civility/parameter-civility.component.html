<app-civility-list-view></app-civility-list-view>
