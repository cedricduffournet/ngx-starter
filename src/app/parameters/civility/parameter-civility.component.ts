import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-civility',
  templateUrl: './parameter-civility.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterCivilityComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
