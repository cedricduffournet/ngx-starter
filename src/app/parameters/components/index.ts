export * from './referenced-table-menu/referenced-table-menu.component';
