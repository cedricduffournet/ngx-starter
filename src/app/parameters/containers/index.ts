export * from './list-parameters/list-parameters.component';
