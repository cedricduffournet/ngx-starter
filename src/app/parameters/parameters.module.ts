import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';
import { PageInnerModule } from '@app/shared/page-inner/page-inner.module';
import { CivilityModule } from '@app/civility/civility.module';

import { ListParametersComponent } from '@app/parameters/containers';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { ReferencedTableMenuComponent } from '@app/parameters/components';

import { routing } from './parameters.routing';

@NgModule({
  imports: [
    CommonModule,
    TranslateModule,
    PageInnerModule,
    CivilityModule,
    routing,
    ReactiveFormsModule
  ],
  declarations: [
    ListParametersComponent,
    ReferencedTableMenuComponent,
    ParameterCivilityComponent
  ]
})
export class ParametersModule {}
