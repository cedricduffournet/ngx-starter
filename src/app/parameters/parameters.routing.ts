import { RouterModule, Routes } from '@angular/router';
import { ListParametersComponent } from '@app/parameters/containers';
import { ParameterCivilityComponent } from '@app/parameters/civility';
import { AuthGuard } from '@app/authentication/services';

export const routes: Routes = [
  {
    pathMatch: 'full',
    path: '',
    component: ListParametersComponent,
    data: { title: 'PARAMETERS' }
  },
  {
    path: 'civilities',
    component: ParameterCivilityComponent,
    canActivate: [AuthGuard],
    data: {
      roles: [
        'ROLE_CIVILITY_CREATE',
        'ROLE_CIVILITY_EDIT',
        'ROLE_CIVILITY_DELETE'
      ],
      title: 'CIVILITIES',
      previousLink: ['parameters']
    }
  }
];
export const routing = RouterModule.forChild(routes);
