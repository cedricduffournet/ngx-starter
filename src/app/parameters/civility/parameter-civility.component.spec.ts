import { Component } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ParameterCivilityComponent } from '@app/parameters/civility';

@Component({ selector: 'app-civility-list-view', template: '' })
class ListCivilityViewStubComponent {}

describe('ParameterCivilityComponent', () => {
  let fixture: ComponentFixture<ParameterCivilityComponent>;
  let component: ParameterCivilityComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ParameterCivilityComponent, ListCivilityViewStubComponent]
    });

    fixture = TestBed.createComponent(ParameterCivilityComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
