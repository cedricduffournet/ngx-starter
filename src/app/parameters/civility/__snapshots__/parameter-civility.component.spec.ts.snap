// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParameterCivilityComponent should be created 1`] = `
<app-parameter-civility>
  <app-civility-list-view />
</app-parameter-civility>
`;
